packages/command-palette/src/index.tsx

```diff
@@ -96,22 +83,19 @@ const StyledCommandsFooter = styled.div( {
 	color: 'var(--studio-gray-50)',
 } );
 
-export function CommandMenuGroup( {
-	currentSiteId,
-	search,
-	close,
-	setSearch,
-	setPlaceholderOverride,
-	selectedCommandName,
-	setSelectedCommandName,
-	setFooterMessage,
-	setEmptyListNotice,
-	navigate,
-	useCommands,
-	currentRoute,
-	useSites,
-	userCapabilities,
-}: CommandMenuGroupProps ) {
+export function CommandMenuGroup() {
+	const {
+		search,
+		close,
+		setSearch,
+		setPlaceholderOverride,
+		selectedCommandName,
+		setSelectedCommandName,
+		setFooterMessage,
+		setEmptyListNotice,
+	} = useCommandMenuGroupContext();
+	const { currentSiteId, navigate, useCommands, currentRoute, useSites, userCapabilities } =
```

Comment:
I think most of these are not needed by this component, but by the `useCommandPalette` hook. Can we move the call to the `useCommandMenuGroupContext` hook inside the `useCommandPalette` hook? Otherwise we still have the same problem of parent components using props that are only needed by children components.

Reply:
Good catch, addressed on https://github.com/Automattic/wp-calypso/pull/88672/commits/a0f0ff18b214af4b9282865030d87d3dcd9707e3